src/System.Text.Primitives/System/Text/InvariantParser_int.cs

```diff
@@ -98,10 +88,18 @@ unsafe public static bool TryParse(byte* utf8Text, int index, int length, out in
             value = 0;
             bytesConsumed = 0;
             bool negative = false;
+            bool signed = false;
 
             if (utf8Text[index] == '-')
             {
                 negative = true;
+                signed = true;
```

Comment:
We only do that if the next character is not a digit. e.g. if the user submits a buffer with "-sdf", we reset the byte counter to zero and return false.